src/test/java/plugins/ForemanNodeSharingPluginTest.java

```diff
@@ -143,7 +144,18 @@ public void testMatrixJob() {
         job.addUserAxis("X", "1 2 3 4 5");
         job.save();
 
-        MatrixBuild b = job.startBuild().waitUntilFinished(360).as(MatrixBuild.class);
+        MatrixBuild b = job.startBuild().as(MatrixBuild.class);
+
+        jenkins.runScript("def items = Jenkins.getInstance().getQueue().getBuildableItems();\n" +
+                "for (hudson.model.Queue$BuildableItem i: items) {\n" +
+                "  Label l = i.getAssignedLabel();\n" +
+                "  if (l != null) {\n" +
+                "    println('suggesting review of label ' + l.getName());\n" +
+                "    l.nodeProvisioner.suggestReviewNow();\n" +
+                "  }\n" +
+                "}");
```

Comment:
I do not really like this intervention, especially if it is not necessary to implement the test. We are getting quite whiteboxy here...